Mathlib/RingTheory/DedekindDomain/AdicValuation.lean

```diff
@@ -571,4 +573,41 @@ lemma adicCompletion.mul_nonZeroDivisor_mem_adicCompletionIntegers (v : HeightOn
     -- and now it's easy
     omega
 
+section AbsoluteValue
+
+open WithZeroMulInt
+
+variable {R K} in
+/-- The `v`-adic absolute value function on `K` defined as `b` raised to negative `v`-adic
+valuation, for some `b` in `ℝ≥0` -/
+def adicAbvDef (v : HeightOneSpectrum R) {b : NNReal} (hb : 1 < b) :=
+  fun x ↦ toNNReal (ne_zero_of_lt hb) (v.valuation K x)
+
+variable {R K} in
+lemma isNonanchimedean_adicAbvDef {b : NNReal} (hb : 1 < b) :
+    IsNonarchimedean (α := K) (fun x ↦ v.adicAbvDef hb x) := by
+  intro x y
+  simp only [adicAbvDef]
+  have h_mono := (toNNReal_strictMono hb).monotone
+  rw [← h_mono.map_max]
+  exact h_mono ((v.valuation _).map_add x y)
+
+variable {R K} in
+/-- The `v`-adic absolute value on `K` defined as `b` raised to negative `v`-adic
+valuation, for some `b` in `ℝ≥0` -/
+def adicAbv (v : HeightOneSpectrum R) {b : NNReal} (hb : 1 < b) : AbsoluteValue K ℝ where
```

Comment:
Can you add this to the "Main definitions" section of the module doc at the beginning of the file?